inlong-dataproxy/dataproxy-source/src/main/java/org/apache/inlong/dataproxy/source/tcp/InlongTcpSource.java

```diff
@@ -0,0 +1,237 @@
+
+            try {
+                if (host == null) {
+                    nettyChannel = serverBootstrap.bind(new InetSocketAddress(port));
+                } else {
+                    nettyChannel = serverBootstrap.bind(new InetSocketAddress(host, port));
+                }
+            } catch (Exception e) {
+                LOG.error("Inlong TCP Source error bind host {} port {},program will exit!", host,
+                        port);
+                System.exit(-1);
+            }
+
+            allChannels.add(nettyChannel);
+
+            LOG.info("Inlong TCP Source started at host {}, port {}", host, port);
+            super.start();
+        } catch (Throwable t) {
+            t.printStackTrace();
+        }
+    }
+
+    /**
+     * stop
+     */
+    @Override
+    public synchronized void stop() {
+        LOG.info("[STOP SOURCE]{} stopping...", super.toString());
+        if (allChannels != null && !allChannels.isEmpty()) {
+            try {
+                allChannels.unbind().awaitUninterruptibly();
+                allChannels.close().awaitUninterruptibly();
+            } catch (Exception e) {
+                LOG.warn("Inlong TCP Source netty server stop ex", e);
+            } finally {
+                allChannels.clear();
+            }
+        }
+
+        if (serverBootstrap != null) {
+            try {
+                serverBootstrap.releaseExternalResources();
+            } catch (Exception e) {
+                LOG.warn("Inlong TCP Source serverBootstrap stop ex ", e);
+            } finally {
+                serverBootstrap = null;
+            }
+        }
+
+//        super.stop();
+        LOG.info("[STOP SOURCE]{} stopped", super.getName());
+    }
+
+    /**
+     * configure
+     * 
+     * @param context
+     */
+    @Override
+    public void configure(Context context) {
+        try {
+            LOG.info("context is {}", context);
+            this.sourceContext = new SourceContext(this, allChannels, context);
+            // start
+            this.sourceContext.start();
+
+            tcpNoDelay = context.getBoolean(ConfigConstants.TCP_NO_DELAY, true);
+            keepAlive = context.getBoolean(ConfigConstants.KEEP_ALIVE, true);
+            highWaterMark = context.getInteger(ConfigConstants.HIGH_WATER_MARK, 64 * 1024);
+            receiveBufferSize = context.getInteger(ConfigConstants.RECEIVE_BUFFER_SIZE, 1024 * 64);
+            if (receiveBufferSize > 16 * 1024 * 1024) {
+                receiveBufferSize = 16 * 1024 * 1024;
+            }
+            Preconditions.checkArgument(receiveBufferSize > 0, "receiveBufferSize must be > 0");
+
+            sendBufferSize = context.getInteger(ConfigConstants.SEND_BUFFER_SIZE, 1024 * 64);
+            if (sendBufferSize > 16 * 1024 * 1024) {
+                sendBufferSize = 16 * 1024 * 1024;
+            }
+            Preconditions.checkArgument(sendBufferSize > 0, "sendBufferSize must be > 0");
```

Comment:
202-206 and 208-212 can be abstracted to a function

Reply:
fixed it.